gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/service/InboundMessageMergingService.java

```diff
@@ -69,19 +68,18 @@ private void findAndReplaceSkeleton(List<PatientAttachmentLog> attachmentLogs, I
             throws SAXException, TransformerException {
         // merge skeleton message into original payload
         var skeletonLogs = attachmentLogs.stream().filter(log -> log.getSkeleton().equals(true)).toList();
-        var skeletonFileName = skeletonLogs.stream().findFirst().get().getFilename();
+        var skeletonFileName = skeletonLogs.stream().findFirst().map(PatientAttachmentLog::getFilename).orElseThrow();
```

Comment:
Do we need to update any unit tests with these modifications?

Reply:
The `map` + `orElseThrow` was introduced to avoid the unchecked `get()`. I've refactored the code so throwing the exception is no longer required.